### DISCUSSION TITLE
How can I run a very large python script?

### DISCUSSION THREAD

**Discussion by LegacyNsfw** [NationalSecurityAgency/ghidra#2608] [Debugger]
I'm trying to convert an IDA Pro project to Ghidra, and my version of IDA Pro is too old for the existing python import/export tools. So I wrote something that converts a .idc script to a series of python commands. It only handles labels and comments, but that will suffice. The generated python script contains almost a megabyte of simple function calls like these:

` createLabel(addressFactory.getAddress("0x440"), "Reset")
` createLabel(addressFactory.getAddress("0x500"), "Operating_System_Checksum")

When I run it via Ghidra's script manager, I get a "module too large" error. The error suggests using compiled python, but I don't found a way to invoke compiled python scripts in Ghidra.

Does anyone have suggestions?

Thanks!

**Comment by astrelsky** [comment]
You can use `toAddr` instead of `addressFactory.getAddress` as it is more convenient. As for shrinking the size of the script you should serialize the data. Json would be quick and easy to use. Using serialized data instead of hardcoding it would also have the benefit of being able to reuse the script with different data.
